osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/tasks/DeviceMessageCleanupJob.java

```diff
@@ -0,0 +1,160 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.core.application.tasks;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.opensmartgridplatform.adapter.domain.core.application.services.TransactionalDeviceLogItemService;
+import org.opensmartgridplatform.logging.domain.entities.DeviceLogItem;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.Assert;
+
+/**
+ * Scheduled job which will find {@link DeviceLogItem} records older than
+ * {@link DeviceMessageCleanupJob#deviceMessageRetentionPeriodInMonths}}. The
+ * old device messages are converted to CSV file and stored to the file system.
+ * Then the old device messages are deleted from the database.
+ */
+@DisallowConcurrentExecution
+public class DeviceMessageCleanupJob extends CsvWriterJob implements Job {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeviceMessageCleanupJob.class);
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.retention}")
+    private int deviceMessageRetentionPeriodInMonths;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.csv.file.location}")
+    private String csvFileLocation;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.csv.file.prefix}")
+    private String csvFilePrefix;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.csv.file.compression.enabled}")
+    private boolean csvFileCompressionEnabled;
+
+    @Value("${osgp.scheduling.job.database.cleanup.device.message.page.size}")
+    private int deviceMessagePageSize;
+
+    @Autowired
+    private TransactionalDeviceLogItemService transactionalDeviceLogItemService;
+
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        LOGGER.info("Quartz triggered cleanup of database - device message records.");
+        final DateTime start = DateTime.now();
+
+        final Date retention = this.calculateDate();
+        final List<DeviceLogItem> oldDeviceMessages = this.transactionalDeviceLogItemService
+                .findDeviceLogItemsBeforeDate(retention, this.deviceMessagePageSize);
+        if (!oldDeviceMessages.isEmpty()) {
+            this.saveDeviceMessagesToCsvFile(oldDeviceMessages);
+
+            LOGGER.info("Deleting device messages...");
+            this.transactionalDeviceLogItemService.deleteDeviceLogItems(oldDeviceMessages);
+
+            System.gc();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Don't try to be smarter than the JVM, remove this call to run the garbage collector. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1215)